pkg/rtc/participant.go

```diff
@@ -2481,10 +2493,24 @@ func (p *ParticipantImpl) handleReceivedDataMessage(kind livekit.DataPacket_Kind
 		if payload.RpcResponse == nil {
 			return
 		}
+
+		rpcResponse := payload.RpcResponse
+		switch res := rpcResponse.Value.(type) {
+		case *livekit.RpcResponse_Payload:
+			shouldForwardData = !p.handleIncomingRpcResponse(payload.RpcResponse.GetRequestId(), res.Payload, nil)
+		case *livekit.RpcResponse_Error:
+			shouldForwardData = !p.handleIncomingRpcResponse(payload.RpcResponse.GetRequestId(), "", &sutils.RpcError{
+				Code:    sutils.RpcErrorCode(res.Error.GetCode()),
+				Message: res.Error.GetMessage(),
+				Data:    res.Error.GetData(),
+			})
+		}
 	case *livekit.DataPacket_RpcAck:
 		if payload.RpcAck == nil {
 			return
 		}
+
+		shouldForwardData = !p.handleIncomingRpcAck(payload.RpcAck.GetRequestId())
```

Comment:
one tricky bit here is an RPC time out clearing that id and a response coming after gets forwarded. Probably going to be okay as clients might just throw it away, but maybe tehre should be a tombstone list of ids and not forward them also. And those can be cleaned up 
- at participant close if the server RPC is not frequent (which should be the case)
- or in some regular go routine (like maybe the susbcriber RTCP worker)

Reply:
It won't really get forwarded anywhere, the destination identity will be set to the same uuid as the RPC ID, SFU's forwarding logic should discard it thinking the participant does not exist.